### DISCUSSION TITLE
Adding error handling for astro-typed-api

### DISCUSSION THREAD

**Discussion by amanketebo** [lilnasy/gratelets#84] [Ideas]
In a previous issue, we talked about adding a error handling api so that it's obvious that error thrown in the fetch method will be handled in another error handler method

@lilnasy You had mentioned something like:
```
defineApiRoute({
    fetch() {},
    throwErrorToClient(errorThrownByFetchAbove) {}
})
```

I think this is a good idea and also maybe throwErrorToClient can be optional too incase the user doesn't want to handle it in any special way. But by making it optional then people might not realize that functionality is there? What do you think @lilnasy?

**Comment by lilnasy** [comment]
I'm guessing the expectation of a user would be that the error data automatically appears on the client side, so they would throw errors. I think the error messages for `ProcedureFailed` (on the server) or `ResponseNotOK` (on the client) can contain a mention (or detailed instructions) on how to send errors to the client in a controlled manner.
